addons/point_of_sale/models/pos_order.py

```diff
@@ -595,6 +620,9 @@ def _default_company_id(self):
         readonly=True,
         states={'draft': [('readonly', False)]},
     )
+    pos_payment_ids = fields.One2many(comodel_name='pos.payment', inverse_name='pos_order_id', string='Pos Payments', readonly=True)
```

Comment:
I'll check onto the removal of statement_ids. I remembered that I considered removing it but I thought it would require to much changes in the frontend. But I'll give it another try to simplify this.